app/lib/DBInterface.scala

```diff
@@ -0,0 +1,215 @@
+package lib
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+import cats.implicits._
+import cats.data.EitherT
+
+
+import play.modules.reactivemongo.ReactiveMongoApi
+import reactivemongo.bson._
+import reactivemongo.api.collections.bson.BSONCollection
+import reactivemongo.api.Cursor
+import reactivemongo.api.commands._
+
+import com.github.nscala_time.time.Imports._
+
+import lib.Data._
+
+
+class DBInterface(api: ReactiveMongoApi) {
+
+  // Getting data from the database----------------------------------------------------------------
+  def getReservations(screeningId: ScreeningId): EitherT[Future, Error, List[Reservation]] = {
```

Comment:
What do you think about extracting trait with all these methods, so we could have different implementations of repository (e.g. storing data in memory instead of real db)?

Reply:
I should have done it. I thought of it myself, unfortunately after I submitted the solution.